src/App.tsx

```diff
@@ -14,11 +14,118 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[],
+  isVisible: boolean,
+  start: boolean,
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: [...goodsFromServer],
+    isVisible: false,
+    start: true,
+  };
+
+  reverseGoods = (goods: string[]) => {
+    return goods.reverse();
```

Comment:
reverse is a mutable method, make a copy before using reverse

Reply:
i disagree. i have had made array copy in state. and mutate it only in one session. when we push "reset" ore "finish" button, we are change mutated array for new copy.